### ISSUE TITLE
Possible discontinuation of QuickReply

### ISSUE THREAD

**Issue by Accio1** [theforumhelpers/QuickReply#16]
I'm not really sure if this is a service that should be maintained here any more. I'm fairly certain that most Forum Helpers don't use it and it goes against the general principle of having unique, constructive posts that we try to encourage in our studio.

We have some options about what we can do with this service:
1. Make this repository private, effectively discontinuing maintaining it.
2. Create a separate overhaul for this repository, which will include coming up with more constructive responses that are used frequently by members of the Forums, as well as improving the usability of this service (#7 is a good example of this, using a dropdown to select Forum names).

@leahcimto thoughts?

**Comment by leahcimto** [comment]
@Accio1 I agree with option 2. This tool has potential, but it really doesn't go well with being constructive like you said.

**Comment by leahcimto** [comment]
@Accio1 Scratch that, I really do think this could be improved. [Here's a comment from dogsmakemehappy on what they think about QuickReply](https://scratch.mit.edu/studios/3688309/comments/#comments-133388400). If QuickReply is discontinued, I do hope this site has something of more use. It could possibly have some tool in relation to applying, or just forming in general. I don't know what that tool would be, but it sounds nice for this site to have more use.

**Comment by FunctionalMetatable** [comment]
~~sorry for popping in/necroposting on an github issue but couldn't you use a json file and change [this](https://scratch.mit.edu/discuss/post/4968917/) code a little bit and you got a simple bookmarklet as QuickReply :D~~
nvm I'm gonna maintain my own version

**Comment by leahcimto** [comment]
@Accio1 ^

**Comment by Accio1** [comment]
@leahcimto https://github.com/theforumhelpers/QuickReply/pull/18#issuecomment-799668009

**Comment by LankyBox01** [comment]
I have an idea, why don't we turn it into an extension?
```
function start(){
var e = document.createElement("div");
e.id="corner"
e.style.width="450px"
e.style.height="15%"
e.style.position="fixed"
e.style.padding="5px"
e.style.top="30px"
e.style.color="white"
e.style.padding="25px"
document.body.appendChild(e)
document.querySelector("#corner").insertAdjacentHTML("afterbegin",`
<font style="font-size: 24px;">Scratch QuickReply Extension</font>`)
}
console.log("Scratch QuickReply Extension activated.");
start()
```

**Comment by Shluffy** [comment]
@Accio1 could I maybe revamp this entire site. 
Here's What I would do:

- You can click a button and it will prompt you multiple times for different things and then will change the textarea to the combined thing
- allow the user to add something custom to the message. This would not be filtered as scratch will filter stuff when posted.
- Make the textarea editable (so they can change from the base without using the prompt method)
- Possibly Use .json files rather then hardcoding stuff into the html

And probably more as it is made. This might take me some time to do, but I can probably do it.

**Comment by Accio1** [comment]
@Shluffy, I was actually considering using Vue.js for the QuickReply overhaul, as a way to see what people think about it, and as kind of a test for https://github.com/theforumhelpers/theforumhelpers.github.io/issues/201 . Each little box could be loaded from a JSON file into a Vue component. Are you familiar with Vue? 

One thing that would be really helpful would be to make a JSON file with the current responses. You could have the forum that it corresponds with, and the text as a part of the JSON file. You could also add places where the user could input information surrounded by {}. For example, in this response: `Hello! This topic is a duplicate of [url=ORIGINAL TOPIC URL]this one over here[/url].`, you could wrap ORIGINAL TOPIC URL in {} to indicate that is is something the user will have to fill in. Would you be interested in handling the JSON file?

**Comment by Shluffy** [comment]
> @Shluffy, I was actually considering using Vue.js for the QuickReply overhaul, as a way to see what people think about it, and as kind of a test for [theforumhelpers/theforumhelpers.github.io#201](https://github.com/theforumhelpers/theforumhelpers.github.io/issues/201) . Each little box could be loaded from a JSON file into a Vue component. Are you familiar with Vue?
> 
> One thing that would be really helpful would be to make a JSON file with the current responses. You could have the forum that it corresponds with, and the text as a part of the JSON file. You could also add places where the user could input information surrounded by {}. For example, in this response: `Hello! This topic is a duplicate of [url=ORIGINAL TOPIC URL]this one over here[/url].`, you could wrap ORIGINAL TOPIC URL in {} to indicate that is is something the user will have to fill in. Would you be interested in handling the JSON file?

@Accio1 no i am not familiar with vue. We could just do `let dupeurl = prompt("What is the url of the topic that this topic is a duplicate of?"); textarea.innerHTML/textarea.innerText = "Hello! This topic is a duplicate of [url=${dupeurl}]this one over here[/url].";` we could also ask for topic id and form the url from that

**Comment by Accio1** [comment]
> > @Shluffy, I was actually considering using Vue.js for the QuickReply overhaul, as a way to see what people think about it, and as kind of a test for [theforumhelpers/theforumhelpers.github.io#201](https://github.com/theforumhelpers/theforumhelpers.github.io/issues/201) . Each little box could be loaded from a JSON file into a Vue component. Are you familiar with Vue?
> > One thing that would be really helpful would be to make a JSON file with the current responses. You could have the forum that it corresponds with, and the text as a part of the JSON file. You could also add places where the user could input information surrounded by {}. For example, in this response: `Hello! This topic is a duplicate of [url=ORIGINAL TOPIC URL]this one over here[/url].`, you could wrap ORIGINAL TOPIC URL in {} to indicate that is is something the user will have to fill in. Would you be interested in handling the JSON file?
> 
> @Accio1 no i am not familiar with vue. We could just do `let dupeurl = prompt("What is the url of the topic that this topic is a duplicate of?"); textarea.innerHTML/textarea.innerText = "Hello! This topic is a duplicate of [url=${dupeurl}]this one over here[/url].";` we could also ask for topic id and form the url from that

Yep, that would work too. It might have to be changed a bit when Vue is added, but it would work in Vanilla JS.

**Comment by FunctionalMetatable** [comment]
I could help

**Comment by Accio1** [comment]
Aside from the JSON file, can we table this until July? I will be on summer break then and will have more time to dedicate to this.

**Comment by Shluffy** [comment]
> Aside from the JSON file, can we table this until July? I will be on summer break then and will have more time to dedicate to this.

@Accio1 fine with me. Also the first step would be to change the backend of quickreply. Could I do that now? this would use variables and pure html injection for the textareas. Or lists

**Comment by gosoccerboy5** [comment]
https://scratch.mit.edu/studios/3688309/comments/#comments-154119589
if the comment chain stops loading after a while, here's a screenshot of the conversation:
<img height="875" src="https://user-images.githubusercontent.com/82768218/123362935-23794600-d537-11eb-8c03-94b3a6c97bc0.png">

**Comment by gosoccerboy5** [comment]
> if the comment chain stops loading after a while, here's a screenshot of the conversation:

the comment chain loads immediately because of the studio update! 😀 

anyways. I feel like quickreply is kind of falling behind the main website and the effort to keep it up isn't worth the value it produces - which seems to not be very much judging by [the comment chain](https://scratch.mit.edu/studios/3688309/comments/#comments-154119589) linked above.
